library/src/main/java/com/alamkanak/weekview/WeekView.java

```diff
@@ -152,20 +156,40 @@ public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float d
             if (mIsZooming)
                 return true;
 
-            // Calculate the direction to scroll.
-            if (mCurrentScrollDirection == Direction.NONE) {
-                // allow scrolling only in one direction
-                if (Math.abs(distanceX) > Math.abs(distanceY)){
-                    mCurrentScrollDirection = Direction.HORIZONTAL;
+            switch (mCurrentScrollDirection) {
+                case NONE: {
+                    // allow scrolling only in one direction
+                    if (Math.abs(distanceX) > Math.abs(distanceY)) {
+                        if (distanceX > 0) {
+                            mCurrentScrollDirection = Direction.LEFT;
+                        } else {
+                            mCurrentScrollDirection = Direction.RIGHT;
+                        }
+                    } else {
+                        mCurrentScrollDirection = Direction.VERTICAL;
+                    }
+                    break;
                 }
-                else {
-                    mCurrentScrollDirection = Direction.VERTICAL;
+                case LEFT: {
+                    // change direction if there was enough change
+                    if (Math.abs(distanceX) > Math.abs(distanceY) && (distanceX < -getWidth() / 40)) {
```

Comment:
Has the number 40 been tested with different number of days selected in the app? Because it seems like a magic value?